core/src/test/java/io/jenkins/pluginhealth/scoring/probes/ThirdPartyRepositoryDetectionProbeTest.java

```diff
@@ -20,21 +29,23 @@ ThirdPartyRepositoryDetectionProbe getSpy() {
     }
 
     @Test
-    void shouldFailIfThirdPartRepositoriesDetected() {
-
+    void shouldFailIfThirdPartRepositoriesDetected() throws IOException, XmlPullParserException {
         final Plugin plugin = mock(Plugin.class);
         final ProbeContext ctx = mock(ProbeContext.class);
-        final ThirdPartyRepositoryDetectionProbe probe = getSpy();
 
-        ClassLoader classLoader = getClass().getClassLoader();
-        File file = new File(classLoader.getResource("fake-test-directory/pom.xml").getFile());
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream("pom-test-both-paths/pom.xml");
+
+        MavenXpp3Reader mavenReader = mock(MavenXpp3Reader.class);
+        Model mockModel = mock(Model.class);
+        List<Repository> repositoryList = mavenReader.read(inputStream).getRepositories();
+        when(mockModel.getRepositories()).thenReturn(repositoryList);
```

Comment:
I am getting `NullPointerException` at line 41

```
java.lang.NullPointerException: Cannot invoke "org.apache.maven.model.Model.getRepositories()" because the return value of "org.apache.maven.model.io.xpp3.MavenXpp3Reader.read(java.io.InputStream)" is null
```

Reply:
You don't want mock the `Model` class. You can to mock the `context.getRepository()` to be pointing at one of the folder that you created in `core/src/test/resources`.